homeassistant/helpers/entity_component.py

```diff
@@ -230,14 +230,15 @@ def async_register_entity_service(
         if isinstance(schema, dict):
             schema = cv.make_entity_service_schema(schema)
 
-        async def handle_service(call: ServiceCall) -> ServiceResponse:
-            """Handle the service."""
-            return await service.entity_service_call(
-                self.hass, self._platforms.values(), func, call, required_features
-            )
-
+        service_func = partial(
+            service.entity_service_call,
+            self.hass,
+            self._platforms.values(),
```

Comment:
If the benefit of the partial is that the arguments are pre-calculated, doesn't pre-calculating `self._platforms.values()` here means the service breaks for any platform added after the service is registered?

Reply:
```
>>> z = {"a":"b","c":"d"}
>>> z.values()
dict_values(['b', 'd'])
>>> view = z.values()
>>> z["e"] = "f"
>>> view 
dict_values(['b', 'd', 'f'])
>>> 
```

Its a view so it should update when the dict changes